nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/listener/RetentionRecordCheckerListener.java

```diff
@@ -38,25 +42,61 @@
 
     @Override
     public void handleEvent(EventBundle events) {
-        for (Event event : events) {
-            checkRecord(event);
+        Map<String, List<String>> docsToCheckAndEvents = new HashMap<String, List<String>>();
+        String ignoreId = null;
+        if (events.containsEventName(RetentionService.RETENTION_CHECKER_LISTENER_IGNORE_EVENT)) {
```

Comment:
the point is to ignore all the events in this bundle that are triggred on the same doc as '
RETENTION_CHECKER_LISTENER_IGNORE_EVENT' => like this we avoid triggering retention on a 'documentModified' notified on the attachRule when the facet was added